File: Anna Park/Week3/2025-06-22-Asteroid Collision.java
import java.util.*;

class asteroidCollision {
    public int[] asteroidCollision(int[] asteroids) {
        
        Stack<Integer> stack = new Stack<>();

        for (Integer a : asteroids){
            boolean alive = true;
            
            while (alive && !stack.isEmpty() && stack.peek()>0 && a<0){
                int b = stack.peek();

                if (-a>b){
                    stack.pop();
                } else if (-a==b){
                    stack.pop();
                    alive = false;
                } else{
                    alive = false;
                }
            }

            if (alive) {
                stack.push(a);
            }
        }

        int[] result = new int[stack.size()];
        for (int i = 0; i < stack.size(); i++) {
            result[i] = stack.get(i);
        }

        return result;
    }
}

File: 2025/shinyoung/2025-06-22-asteroid-collision.py
def asteroidCollision(asteroids):
    stack = []
    for a in asteroids:
        alive = True
        while alive and stack and stack[-1] > 0 and a < 0:
            if stack[-1] < -a:
                stack.pop()
                continue
            elif stack[-1] == -a:
                stack.pop()
            alive = False
        if alive:
            stack.append(a)
    return stack


print(asteroidCollision([5, 10, -5]))
print(asteroidCollision([8, -8]))
print(asteroidCollision([10, 2, -5]))
